components/tracker/src/tls.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    cell::Cell,
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use pin_project::pin_project;
+
+use crate::{
+    slab::{TrackerToken, INVALID_TRACKER_TOKEN},
+    MetricType, Tracker, TrackerTimer, GLOBAL_TRACKERS,
+};
+
+thread_local! {
+    static TLS_TRACKER_TOKEN: Cell<TrackerToken> = Cell::new(INVALID_TRACKER_TOKEN);
+}
+
+pub fn set_tls_tracker(token: TrackerToken) {
+    TLS_TRACKER_TOKEN.with(|c| c.set(token))
+}
+
+pub fn unset_tls_tracker() {
+    TLS_TRACKER_TOKEN.with(|c| c.set(INVALID_TRACKER_TOKEN))
+}
+
+pub fn get_tls_tracker() -> TrackerToken {
+    TLS_TRACKER_TOKEN.with(|c| c.get())
```

Comment:
Can we add an assertion like `TLS_TRACKER_TOKEN.with(|c| c.get()) != INVALID_TRACKER_TOKEN` in debug mode?

Reply:
It's expected that `TLS_TRACKER_TOKEN` can be `INVALID_TRACKER_TOKEN`. To control the total memory usage, we only have limited active trackers. So, it is possible for a request to fail to create a tracker and then the token for this request is `INVALID_TRACKER_TOKEN`. But don't worry, it should not happen in normal cases.

Reply:
A counter is now added.